shared/common-adapters/oriented-image.desktop.js

```diff
@@ -102,79 +164,100 @@ class OrientedImage extends React.Component<Props, State> {
+      const tags = EXIF.readFromBinaryFile(data.buffer)
+      tags ? resolve(tags['Orientation']) : reject(new Error('EXIF failed to read exif data'))
     })
   }
 
-  _handleImageLoadSuccess = (src, orientation) => {
+  _handleOrientationSuccess = orientation => {
     if (!this._hasComponentMounted) return
-    this._setOrientation(src, orientation)
+
+    // If there is no Orientation data set for the image, then mark it as null
+    // in the cache to avoid subsequent calls to EXIF
+    if (!orientation || !isNumber(orientation)) {
+      return this._handleOrientationFailure()
+    }
+
+    this._canvasImageTransform(orientation)
   }
 
-  // Don't perform transforms if the image cannot be loaded or there is no EXIF data
-  _handleImgeLoadFailure = src => {
-    _cacheStyleTransforms[src] = NO_TRANSFORM
+  // Mark this image path as no transform and set the ImageComponent src to the original source
+  _handleOrientationFailure = () => {
+    _cacheNoTransforms[this.props.src] = NO_TRANSFORM
+    this.setState({srcTransformed: this.props.src})
   }
 
-  _setTranformForExifOrientation(src) {
+  _setTranformForExifOrientation = () => {
     if (!this._hasComponentMounted) return
 
     // This image either cannot be transofrmed or does not have an EXIF orientation flag
-    if (_cacheStyleTransforms[src] === NO_TRANSFORM) {
-      return
-    }
-
-    if (_cacheStyleTransforms[src]) {
-      return this.setState({styleTransform: _cacheStyleTransforms[src]})
+    if (_cacheNoTransforms[this.props.src] === NO_TRANSFORM) {
+      this._handleOrientationFailure()
     }
 
     // Uploaded file served from Keybase service
-    if (uploadedSrc.test(src)) {
-      this._fetchExifUploaded(src)
-        .then(orientation => this._handleImageLoadSuccess(src, orientation))
-        .catch(() => this._handleImgeLoadFailure(src))
+    if (uploadedSrc.test(this.props.src)) {
+      this._fetchExifUploaded()
+        .then(orientation => this._handleOrientationSuccess(orientation))
+        .catch(this._handleOrientationFailure)
     } else {
-      this._readExifLocal(src)
-        .then(orientation => this._handleImageLoadSuccess(src, orientation))
-        .catch(() => this._handleImgeLoadFailure(src))
+      this._readExifLocal()
+        .then(orientation => this._handleOrientationSuccess(orientation))
+        .catch(this._handleOrientationFailure)
     }
   }
 
+  /*
+   * Lifecyle Hooks
+   */
   componentDidMount() {
     this._hasComponentMounted = true
-    this._setTranformForExifOrientation(this.props.src)
+    if (this._canvasRef) {
+      this._context = this._canvasRef.getContext('2d')
+      this._setTranformForExifOrientation()
+    }
   }
 
   componentWillUnmount() {
     this._hasComponentMounted = false
+    this._canvasRef = null
+    this._context = null
   }
 
   componentDidUpdate(prevProps: Props, prevState: State) {
     // New src requires changing EXIF transform
     if (prevProps && prevProps.src !== this.props.src) {
```

Comment:
can `prevProps` be falsy?

Reply:
🤦‍♂️ It can't, guaranteed to be called after first render. Thanks :)